### ISSUE TITLE
appgyver logo on iOS ad hoc installation

### ISSUE THREAD

**Issue by osusoy** [AppGyver/steroids#696]
have been away for a while and trying to recreate an old build using the 4.0.0 client. in iOS settings, i have uploaded new custom images everywhere they exist but it seems like something is missing. neither of the following happened with my previous build of the same code using 3.5.3-rc3:
- testflight displays the appgyver icon instead of the app icon
- while installing an ad hoc build, the appgyver icon is displayed but once the installation is finished, it is replaced by the app icon

any ideas?

**Comment by osusoy** [comment]
nothing? just tried a build with 4.0.3-edge3 and the issues is still there. i have provided a list of icons in the ipa, can someone at least confirm whether the list is complete?

**Comment by pre** [comment]
The list of icons you provided is correct. There are no other icons. 

Is the problem you're facing: "The app icon is not being displayed until install has finished. " ?

When you install from Testflight, you will see the correct app icon once the installation is completed. The placeholder icon comes either from iOS or Testflight, it is not submitted by AppGyver. However, this is expected behaviour and not a bug.

I'm still having hard times trying to understand what is the problem here. If you could post a screenshot or visualise it somehow, I could probably understand. If the issue is something else than what I wrote above, I'm confused.

**Comment by osusoy** [comment]
I think the confusion is simply about what is expected and what is a bug but screen shot attached. I suspect it is the same as the issue you refer to. Testflight is being positioned as the main iOS testing platform by Apple and expected behavior is for Steroids apps to be missing an asset it needs? As I mentioned earlier, 3.5.3-rc3 and below displays the correct icon (see custom 3.1.6 scanner icon in the screenshot), 4.0.0 displays the Appgyver icon and anything newer seems to be simply missing whatever icon it is looking for so i would say this is a case of Steroids going backwards. Is there any confirmation that the missing asset won't get in the way of a store submission?

![iconbug](https://cloud.githubusercontent.com/assets/535276/6101111/50af90ca-b072-11e4-9452-b20cacfdaedc.png)

**Comment by osusoy** [comment]
exact same problem using diawi for testing as well.

**Comment by pre** [comment]
That is not an AppGyver logo, but a TestFlight logo. 

Now when you posted this screenshot I finally understood your issue. When we introduced support for iPhone6 and iPhone6+ assets, we removed all the deprecated iOS6 icons. It seems that TestFlight is using one of those for the app icon, therefore, the custom icon does not exist and a TestFlight placeholder icon is displayed instead. 

This is an issue with TestFlight. While we could return the deprecated icons, I think we're not going to do that, instead we'll wait for TestFlight to stop using the deprecated icon sizes. This will probably happen with TestFlight2 which is currently in beta -- have you tried that? 

BOTTOM LINE:
This is a TestFlight placeholder icon. It will NOT be displayed when your application is installed from the Apple App Store.

**Comment by osusoy** [comment]
"exact same problem using diawi for testing as well."
"4.0.0 displays the Appgyver icon and anything newer seems to be simply missing whatever icon it is looking for" - screenshot displays what newer versions do.

bit pointless reporting problems if no-one can be bothered even reading the comments properly. thanks for confirming it was a smart move to dump steroids.

**Comment by pre** [comment]
@osusoy I'm sorry for the disappointment. I just want to let you know that it was unclear for me what the problem was. If you read the problem description now again, you can see that it also lead into wrong direction. 

More time wasn't spent investigating this problem, because the wrong icon was visible only to TestFlight users and did not / will not affect installations from Apple App Store. The problem will solve itself when the new Apple-branded TestFlight will launch -- the current TestFlight service seem to have ceased development in favor of the new platform and therefore this kind of issues pop up.

Sorry to hear about your Steroids experience.

All the best & have nice spring . :)

**Comment by pre** [comment]
We have confirmed that this issue is fixed in the new TestFlight in itunesconnect.apple.com.

The old TestFlight (testflightapp.com) is shutting down tomorrow 26th Februrary, so this issue is no longer a problem for anyone.
